core-profile-tck/tck/src/main/java/ee/jakarta/tck/core/rest/jsonb/cdi/KeysProducer.java

```diff
@@ -33,22 +36,26 @@ public class KeysProducer {
      * @throws Exception - on failure to read or initialize public key
      */
     @PostConstruct
-    private void loadKeys() throws Exception {
-        byte[] pubKeyData;
-        try (InputStream keyIS = getClass().getResourceAsStream("/key.pub")) {
-            if (keyIS == null) {
-                throw new IllegalStateException("Failed to find /key.pub");
+    private void loadKeys() {
+        try {
+            byte[] pubKeyData;
+            try (InputStream keyIS = getClass().getResourceAsStream("/key.pub")) {
+                if (keyIS == null) {
+                    throw new IllegalStateException("Failed to find /key.pub");
+                }
+                pubKeyData = keyIS.readAllBytes();
             }
-            pubKeyData = keyIS.readAllBytes();
+            String pubKeyString = new String(pubKeyData, StandardCharsets.UTF_8);
+            System.out.println(pubKeyString);
+            byte[] keyData = Base64.getDecoder().decode(pubKeyString);
+            EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(keyData);
+            System.out.println(publicKeySpec);
+
+            KeyFactory keyFactory = KeyFactory.getInstance("EC");
+            publicKey = keyFactory.generatePublic(publicKeySpec);
+        } catch (IOException | NoSuchAlgorithmException | InvalidKeySpecException exception) {
+            System.out.println(exception.getStackTrace());
```

Comment:
You can still throw IllegalStateException, which is probably better than just logging just stacktrace and pretend everything is fine. Runtime exception can be undeclared.